### ISSUE TITLE
samba need permission question

### ISSUE THREAD

**Issue by Cr0iX** [yadayada/acd_cli#540]
Hello,
I cannot get it to work with samba, when i try to create or delete a folder it works, but when i try to copy a file to the share it says permission denied?

I've mounted the drive with this:
acd_cli mount -ao --umask 000 --gid groupid --uid userid /dir/mountpoint

Btw. it works on the encrypted drive, but i would like some files not to be encrypted..

Regards
Cr0iX

Love your work!

**Comment by madmanali93** [comment]
do you get a permission denied when you copy a file into it without samba? Or is it strictly happening in samba?

**Comment by Cr0iX** [comment]
madmanali93:
It only happens in samba.
This is my samba share config:

[Amazon-Encrypted]
       comment = amazondrive
       path = /amazon-drive/encrypted-mountpoint/
       public = no
       writable = yes
       browseable = yes

**Comment by madmanali93** [comment]
are you able to copy anything if you change the path to a non acd folder?

**Comment by Cr0iX** [comment]
Yes i have serveral other shared folders and none of them have any problems.

**Comment by MrColdbird** [comment]
[global]
server string = YOURSERVERNAME
workgroup = WORKGROUP
netbios name = %h
guest account = amazon
socket options = TCP_NODELAY IPTOS_LOWDELAY SO_RCVBUF=65536 SO_SNDBUF=65536
smb ports = 445
max protocol = SMB2
min receivefile size = 16384
deadtime = 30
os level = 20
mangled names = no
syslog only = yes
syslog = 2
name resolve order = lmhosts wins bcast host
preferred master = auto
domain master = auto
local master = yes
printcap name = /dev/null
load printers = no
browseable = yes
writeable = yes
printable = no
encrypt passwords = true
enable core files = no
passdb backend = smbpasswd
smb encrypt = disabled
use sendfile = yes
security = user
map to guest = Bad User

[amazon]
comment = Amazon Drive encfs
path = /your/encfs/mountpoint
available = yes
browsable = yes
writable = yes
public = yes
guest ok = yes

---------------------------------

Adjust the share path and servername as required.

The settings I've used also help in stabilizing fuse-fs uploads a bit, anything up to 500mb filesize should upload fine with those settings, beyond that you will get "network timeout errors" reported by windows.

Make sure to run your amazon and encfs mount under a newly created service user, I personally use a systemctl service script to pull up my mounts and keep them mounted, in my case this user is called "amazon".

By mapping the "Bad User", aka. guest account to this amazon linux user, it also takes care of any permission issues you might be facing while using the share.

Optionally, you might also want to turn on allow_other in the fuse settings, so that you can mount fuse filesystems in a better fashion.
